.eleventy.js

```diff
@@ -1,7 +1,141 @@
+		.use(require("markdown-it-footnote"))
+
+		// markdown-it-mark lets you use `==marked==` to highlight text.
+		.use(require("markdown-it-mark"))
+
+		// markdown-it-container lets you easily create callouts and other kinds of simple containers in markdown..
+		//  We configure it here to support stuff like: `:::tip\nthis is in the tip div\n:::\n`.
+		//  It needs to be accompanied by css to highlight `.callout-tip` (or the other keywords we used here).
+		.use(require("markdown-it-container"), "warn", markdownItContainerCfg("warn"))
+		.use(require("markdown-it-container"), "info", markdownItContainerCfg("info"))
+		.use(require("markdown-it-container"), "tip",  markdownItContainerCfg("tip"))
+		.use(require("markdown-it-container"), "todo", markdownItContainerCfg("todo"))
+
+		// You may also want to consider yet more plugins, like https://github.com/markdown-it/markdown-it-emoji .
+	)
+
+	// This navigation plugin consumes frontmatter from each page,
+	//  and gathers that info into a form that we use to build the nav menu.
+	//   You can see an example of this used as global navigation here: https://github.com/ipld/ipld/blob/2f010292da7e3d089f2512b175494abe114f1a2a/.site/_layouts/main.njk#L33-L53
+	//   You can see an example of a smaller local insertable here: https://github.com/ipld/ipld/blob/2f010292da7e3d089f2512b175494abe114f1a2a/.site/_includes/listing.njk
+	//   In some parts of the above examples, this data configuration was also entangled: https://github.com/ipld/ipld/blob/2f010292da7e3d089f2512b175494abe114f1a2a/.site/_data/eleventyComputed.js#L4-L11
+	//   (FUTURE: consider if we can bring those applications together in one place here by using `eleventyConfig.addShortcode` or `eleventyConfig.addFilter` or similar?)
+	eleventyConfig.addPlugin(require("@11ty/eleventy-navigation"))
+
+	// Make some functions available as filters -- which means they can be used in nunjucks.
+	function() {
+		// Make path manipulation available, so we can use them to create breadcrumbs and other navigation elements.
+		//  You can see these used in breadcrumb production here: https://github.com/ipld/ipld/blob/2f010292da7e3d089f2512b175494abe114f1a2a/.site/_layouts/main.njk#L20-L30
+		const { dirname, basename } = require("path")
+		eleventyConfig.addFilter("dirname", dirname)
+		eleventyConfig.addFilter("basename", basename)
+
+		// Make build time available, so we can use it to make sure assets like CSS aren't cached beyond reason.
+		//  (This approach is a bit dumb, and will bust caches more than necessary, but is much simpler to implement than content hashing.)
+		//  You can invoke this like for example `<link rel="stylesheet" href="/css/style.css?{{ "" | cachebuster }}">` in a njk template.
+		const buildtime = new Date().getTime()
+		eleventyConfig.addFilter("cachebuster", () => buildtime)
+	}()
+
+	// Here are all the additional files and folders unique to this site.
```

Comment:
Nice! We could also lift this out of the main function and name it to make it obvious what it's used for, but I guess it works with adding a comment on top like you've done as well.

Reply:
Agree, I think that could be a useful subsequent step.